### ISSUE TITLE
Cannot statically compile on Windows due to linking failure, exit code 1120

### ISSUE THREAD

**Issue by atlanticaccent** [OSSystems/compress-tools-rs#57]
Attempting to compile my project or examples statically fails with error `error: linking with 'link.exe' failed: exit code: 1120`.

The following is also provided:
```
note:    Creating library C:\Users\wanty\Documents\compress-tools-rs-master\target\debug\examples\uncompress_tool.lib and object C:\Users\wanty\Documents\compress-tools-rs-master\target\debug\examples\uncompress_tool.exp
          archive.lib(filter_fork_windows.c.obj) : error LNK2019: unresolved external symbol __imp_WaitForInputIdle referenced in function __archive_create_child
          libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_GetProcessWindowStation referenced in function OPENSSL_isservice
          libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_GetUserObjectInformationW referenced in function OPENSSL_isservice
          libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_MessageBoxW referenced in function OPENSSL_showfatal
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertOpenStore referenced in function capi_list_certs
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertCloseStore referenced in function capi_find_key
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertEnumCertificatesInStore referenced in function capi_find_cert
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertFindCertificateInStore referenced in function capi_find_cert
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertDuplicateCertificateContext referenced in function capi_load_ssl_client_cert
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertFreeCertificateContext referenced in function capi_dsa_free
          libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertGetCertificateContextProperty referenced in function capi_cert_get_fname
          C:\Users\wanty\Documents\compress-tools-rs-master\target\debug\examples\uncompress_tool.exe : fatal error LNK1120: 11 unresolved externals
```

This seems to be similar to previous issue #47.

Steps to reproduce:

- Install libarchive with vcpkg using tuple "x64-windows-static"
- Either set environment variable RUSTFLAGS='-C target-feature=+crt-static' OR add the following to the config file in the .cargo directory: 
```
[target.x86_64-pc-windows-msvc]
rustflags = ["-C", "target-feature=+crt-static"]
```
- The VCPKG_DYNAMIC environment variable must not be set/must be set to 0
- (Possibly) Set the environment variable OPENSSL_ROOT_DIR="{vcpkg install directory}\installed\x64-windows-static"
- Attempt to build compress-tools examples

**Comment by otavio** [comment]
@atlanticaccent, this is a known issue, but honestly, I don't know how to fix it. Any idea how to address it?

**Comment by atlanticaccent** [comment]
Unfortunately I'm not terribly versed in any of this. I can confirm that it is possible to compile statically for linux by setting the appropriate RUSTFLAGS and targeting x86_64-unknown-linux-musl, if that helps at all.

I'm curious if there's another way to make OpenSSL available to the compiler, maybe through an openssl crate, but I don't really know how you'd properly go about that.

**Comment by otavio** [comment]
I believe it has a `vendored` feature flag. I'd love to have someone more versed on Windows (and `vcpkg`) to help here.

**Comment by TcNobo** [comment]
Was Googling for "OPENSSL_isservice" as I'm having 3 of the same errors as you, and came across this page.
Came across this [tutorial here](https://www.programmersought.com/article/8711142745/), and even though it's not in English I see them adding `User32.lib`. 
After adding `#pragma comment (lib, "User32.lib")` the issues I was having disappeared.

I now have: Normaliz.lib, Ws2_32.lib, Wldap32.lib, Crypt32.lib, advapi32.lib and User32.lib

**Comment by atlanticaccent** [comment]
No idea if that's the issue or not here, but this inspired me to do some googling again. [https://stackoverflow.com/questions/37522654/linking-with-openssl-lib-statically](This), [https://github.com/chronoxor/CppServer/issues/10](this) and [http://openssl.6102.n7.nabble.com/openssl-1-1-0-Linker-error-on-Windows-td66318.html](this) all refer to some Windows `crypt32.lib` and `crypt.lib`.

Suggestions include adding:
- `#pragma comment (lib, "crypt32");` but I don't know where
- configure `capieng` with `enable-capieng`
- chuck crypt or crypt32 into the path variables (this may be a MVSC/Visual Studio specific solution)
- add `/Lcrypt32` linker flag, somewhere
- somehow disable cpieng entirely by > use `no-capieng` Configure option

Sorry that I can't elaborate further than that, all of this is beyond me. Crypt/Crypt32 do seem like very promising leads however.

**Comment by otavio** [comment]
Could you guys help on this issue?

**Comment by otavio** [comment]
Fixed in #69.

**Comment by dontlooksoworried** [comment]
> Was Googling for "OPENSSL_isservice" as I'm having 3 of the same errors as you, and came across this page. Came across this [tutorial here](https://www.programmersought.com/article/8711142745/), and even though it's not in English I see them adding `User32.lib`. After adding `#pragma comment (lib, "User32.lib")` the issues I was having disappeared.
> 
> I now have: Normaliz.lib, Ws2_32.lib, Wldap32.lib, Crypt32.lib, advapi32.lib and User32.lib

Wow, this solved my issue unrelated to this project. I had been searching similar errors through many posts and could not find a viable solution. For some reason this include instantly solved all my issues as well. Now to investigate further... Thanks!
